content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md

```diff
@@ -0,0 +1,67 @@
++++
+# `title` is how your post will be listed and what will appear at the top of the post
+title=  "AWS to Contribute Efficiency Improvements to Valkey 8 "
+# `date` is when your post will be published.
+# For the most part, you can leave this as the day you _started_ the post.
+# The maintainers will update this value before publishing
+# The time is generally irrelvant in how Valkey published, so '01:01:01' is a good placeholder
+date= 2024-07-08 01:01:01
+# 'description' is what is shown as a snippet/summary in various contexts.
+# You can make this the first few lines of the post or (better) a hook for readers.
+# Aim for 2 short sentences.
+description= "Better efficiency reduces cost, improves latency and makes our environment greener"
+# 'authors' are the folks who wrote or contributed to the post.
+# Each author corresponds to a biography file (more info later in this document)
+authors= [ "dantouitou", "uriyagelnik"]
++++
+## AWS to Contribute Efficiency Improvements for Valkey 8
+
+From simple in-memory caching implementations to complex job queues, real-time collaboration, and leaderboards applications, we at AWS are continually amazed by how innovatively users employ Valkey. 
+
+Clearly, this is just the tip of the iceberg. As more use cases and industries want to  benefit from the speed, low latency, and cost reduction advantages of in-memory processing as introduced by Valkey, we are fully committed to this vision.
+
+### Our Commitment to Efficiency
+
+One of our primary goals at AWS is to ensure our customers receive the most efficient services. Efficiency not only leads to lower costs, better latency and a greener environment, but also enhances resilience.
+
+For example, take the fork-based snapshotting and replication process. The longer it takes, the more memory is consumed by copied-on-write pages and the replica output buffer, often resulting in high swapping and erratic behavior. By reducing execution time, one can achieve a more predictable memory usage and greater stability.
+
+### Major Upgrade to Valkey Performance
+
+We are excited about the new Linux Foundation sponsorship for Valkey and are taking a bigger step, contributing our major performance improvements and expertise. Starting with version 8, Valkey users will benefit from a breakthrough in performance, thanks to a new multi-threading implementation that can can considerably boost performance on a wide range of hardware types. 
+
+For caching workloads, Valkey users will be able to increase maximum requests per second to over 1 million on multi-core machines such as AWS EC2  r7g.4xl
```

Comment:
We need to couch this statement as well, because in OSS we talk a lot about pipeline performance, and Valkey can already do 1M+ RPS per process with batching.

Reply:
I suggest running the same benchmark, same hardware, with and without these patches to to show the difference. (Or compare to Valkey 7.2.5 to include other optimizations too, but then with some attribution to others too.)